src/preferences/broadcastsettings.cpp

```diff
@@ -97,26 +97,22 @@ bool BroadcastSettings::addProfile(BroadcastProfilePtr profile) {
             &BroadcastProfile::connectionStatusChanged,
             this,
             &BroadcastSettings::onConnectionStatusChanged);
-    m_profiles.insert(profile->getProfileName(), BroadcastProfilePtr(profile));
+    m_profiles.insert(profile->getProfileName(), profile);
 
     emit profileAdded(profile);
     return true;
 }
 
-bool BroadcastSettings::saveProfile(BroadcastProfilePtr profile) {
-    if (!profile) {
-        return false;
-    }
-
-    QString filename = profile->getLastFilename();
+bool BroadcastSettings::saveProfile(BroadcastProfile& profile) {
```

Comment:
```suggestion
bool BroadcastSettings::saveProfile(const BroadcastProfile& profile) {
```
I guess that will work, if you make some BroadcastProfile functions const as well. Otherwise use a plain pointer, this is anyway the preferred style for functions that promise not to store the pointer.

Reply:
Force pushed https://github.com/mixxxdj/mixxx/pull/4751/commits/a9f65b7b979fa60cc24f3f2e8ee2cc957aeb5099 using `BroadcastProfile*` instead of `BroadcastProfile&` and undoing some of the previous changes.

Reply:
Force pushed https://github.com/mixxxdj/mixxx/pull/4751/commits/c898ed9480a5b79eeab4149833e3a85e4666ec45 to fix the build on Ubuntu 18.04.